synapse/storage/databases/main/__init__.py

```diff
@@ -516,11 +517,14 @@ def get_users_paginate(
 
         def get_users_paginate_txn(txn):
             filters = []
-            args = []
+            args = [self.hs.config.server_name]
 
             if name:
+                filters.append("(name LIKE ? OR displayname LIKE ?)")
+                args.extend(["%" + name + "%:%", "%" + name + "%"])
```

Comment:
```suggestion
                args.extend(["%@" + name + "%:%", "%" + name + "%"])
```

Otherwise the `@` will be included as part of the search.

Reply:
Should have been the other way round: `"@% + name + "%..."`